### PR TITLE
[Network modifications] Allow to create injections and branches with bay in BUS_BREAKER topology

### PR DESCRIPTION
**Pull request by colineplqt**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
CreateFeederBay and CreateBranchFeeder wrap to support creation in bus/breaker voltage levels.

### CODE REVIEW

File: pytest.ini [hunk 1]
```diff
@@ -1,4 +1,4 @@
 [pytest]
 #Change those options to get logs
 log_cli=false
-log_level=ERROR
+log_level=DEBUG
```
Comment (EtienneLt) [inline pytest.ini]: I think you forget to revert it before pushing

File: tests/test_network_modification.py [hunk 2]
```diff
@@ -6,6 +6,9 @@
 
 TEST_DIR = pathlib.Path(__file__).parent
 
+import logging
+logging.basicConfig()
+logging.getLogger('powsybl').setLevel(1)
```
Comment (EtienneLt) [inline tests/test_network_modification.py]: same

File: java/src/main/java/com/powsybl/dataframe/network/adders/NetworkUtils.java [hunk 3]
```diff
@@ -28,6 +26,30 @@ public static VoltageLevel getVoltageLevelOrThrow(Network network, String id) {
         return voltageLevel;
     }
 
+    public static VoltageLevel getVoltageLevelOrThrowWithBusOrBusbarSectionId(Network network, int row, StringSeries voltageLevels, StringSeries busOrBusbarSections) {
+        VoltageLevel voltageLevel;
```
Comment (EtienneLt) [inline java/src/main/java/com/powsybl/dataframe/network/adders/NetworkUtils.java]: not important you can delete this line and return directly the voltage level in the "if" sections

File: java/src/main/java/com/powsybl/dataframe/network/adders/NetworkUtils.java [hunk 4]
```diff
@@ -28,6 +26,28 @@ public static VoltageLevel getVoltageLevelOrThrow(Network network, String id) {
...
+                }
+                if (busOrBusbarSection instanceof BusbarSection) {
+                    return ((BusbarSection) busOrBusbarSection).getTerminal().getVoltageLevel();
+                } else if (busOrBusbarSection instanceof Bus) {
+                    return ((Bus) busOrBusbarSection).getVoltageLevel();
+                } else {
+                    throw new PowsyblException(String.format("Unsupported type %s for identifiable %s", busOrBusbarSection.getType(), busOrBusbarSection.getId()));
+                }
+            } else {
+                throw new PowsyblException("voltage_level_id is missing");
```
Comment (EtienneLt) [inline java/src/main/java/com/powsybl/dataframe/network/adders/NetworkUtils.java]: should it be more explicite like voltage level or bus id or busbar section id is missing ?

File: java/src/main/java/com/powsybl/dataframe/network/adders/NetworkUtils.java [hunk 5]
```diff
@@ -28,6 +26,28 @@ public static VoltageLevel getVoltageLevelOrThrow(Network network, String id) {
         return voltageLevel;
     }
 
+    public static VoltageLevel getVoltageLevelOrThrowWithBusOrBusbarSectionId(Network network, int row, StringSeries voltageLevels, StringSeries busOrBusbarSections) {
+        if (voltageLevels == null) {
+            if (busOrBusbarSections.get(row) != null) {
```
Comment (EtienneLt) [inline java/src/main/java/com/powsybl/dataframe/network/adders/NetworkUtils.java]: I think busOrBusbarSections can be null

File: tests/test_network_modification.py [hunk 6]
```diff
@@ -183,12 +171,22 @@ def test_add_load_bay_from_kwargs():
     assert position.direction == 'BOTTOM'
 
 
+def test_add_load_bay_from_kwargs_bus_breaker():
```
Comment (EtienneLt) [inline tests/test_network_modification.py:174]: should it be in test_network_modification ?

Comment (colineplqt) [reply tests/test_network_modification.py:174]: Why not?

Comment (EtienneLt) [reply tests/test_network_modification.py:174]: my mistake I thought it was in test_network

**Review by EtienneLt — APPROVED** [review]

**Pull request merged**